libbpfgo/libbpfgo.go

```diff
@@ -651,3 +653,28 @@ func (pb *PerfBuffer) poll() error {
 	}
 	return nil
 }
+
+// TracePrint reads data from the trace pipe that bpf_trace_printk() writes to,
+// and writes it to stdout. The pipe is global, so this function is not
+// associated with any BPF program. It is recommended to use bpf_trace_printk()
+// and this function for debug purposes only.
+// This is a blocking function intended to be called from a goroutine, for example:
+//		go libbpfgo.TracePrint()
+func TracePrint() {
+	f, err := os.Open("/sys/kernel/debug/tracing/trace_pipe")
+	if err != nil {
+		fmt.Println("TracePrint failed to open trace pipe: %v", err)
```

Comment:
Imo this should be printed to stderr instead

Reply:
or return an error instead of printing it

Reply:
Can't use the return value if called from a goroutine. Could send it back in a channel, but that seems like overkill given this is really only suitable for very basic debug purposes (where you're probably logging to screen anyway). Will send it to stderr though, that's a good point.

Reply:
this one (unlike L674) can be returned, as it's determined before entering the loop

Reply:
What I mean is the caller can't use the return value if they call it from a goroutine, which (per the doc comment) they would almost certainly want to do because it's blocking

Reply:
oh, I assumed you will take my suggestion to own the goroutine, in that case it would attempt to open the file before starting the goroutine